src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -330,7 +330,7 @@ private static function parseHostPort(array $url, array &$info): array
      */
     private static function dnsResolve(string $host, NativeClientState $multi, array &$info, ?\Closure $onProgress): string
     {
-        $flag = '' !== $host && '[' === $host[0] && ']' === $host[-1] && str_contains($host, ':') ? \FILTER_FLAG_IPV6 : \FILTER_FLAG_IPV4;
+        $flag = '' !== $host && str_starts_with($host, '[') && str_ends_with($host, ']') && str_contains($host, ':') ? \FILTER_FLAG_IPV6 : \FILTER_FLAG_IPV4;
```

Comment:
let's revert these changes, consistency is not a concern here and saving needless function calls is desired